src/pkgcheck/checks/reserved.py

```diff
@@ -0,0 +1,100 @@
+import re
+from .. import addons, bash, results, sources
+from . import Check
+
+
+reserved_prefixes = ('__', 'abort', 'dyn', 'prep')
+reserved_substrings = ('hook', 'paludis')  # 'ebuild' is special case
+reserved_ebuild_regex = re.compile(r'.*[^a-zA-Z]ebuild.*')
```

Comment:
AFAICS this will catch `ebuild` prefixed by a non-letter but it won't catch `ebuild` at beginning of string.